src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  let parts = sourceString.split(';');
+
+  parts = parts.filter((part) => part.trim() !== '');
```

Comment:
Instead of splitting and then filtering the array in separate steps, consider chaining these methods for better readability.